editor/src/components/canvas/canvas-strategies/strategies/reparent-helpers/reparent-helpers.ts

```diff
@@ -56,6 +76,74 @@ export function isAllowedToReparent(
   }
 }
 
+export function canCopyElement(
+  editor: EditorState,
+  target: ElementPath,
+): Either<string, ElementPath> {
+  const metadata = MetadataUtils.findElementByElementPath(editor.jsxMetadata, target)
+  if (MetadataUtils.isElementGenerated(target)) {
+    return left('Cannot copy generated element')
+  }
+
+  if (metadata == null) {
+    const parentPath = EP.parentPath(target)
+    const conditional = findMaybeConditionalExpression(parentPath, editor.jsxMetadata)
+    if (conditional != null) {
+      const branchCase = maybeBranchConditionalCase(parentPath, conditional, target)
+      if (branchCase == null) {
+        return left('Cannot copy empty branch')
+      }
+      return right(target)
+    }
+    return left('Cannot find element metadata')
+  }
+
+  return foldEither(
+    (_) => right(target),
+    () => {
+      if (!MetadataUtils.targetHonoursPropsPosition(editor.projectContents, metadata)) {
+        return left('target does not honour positioning props')
```

Comment:
I think we should allow copying an element and on paste maybe we can't position it but it's still pasted

Reply:
great idea! removed it in [9503dfd](https://github.com/concrete-utopia/utopia/pull/3778/commits/9503dfd781b11a8a348357b305e752aad836deb0)